hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/records/CryptoDeleteRecordBuilder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl.records;
+
+import com.hedera.hapi.node.base.AccountID;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Map;
+
+/**
+ * A {@code RecordBuilder} specialization for tracking the side effects of a {@code CryptoDelete}
+ * transaction.
+ */
+public interface CryptoDeleteRecordBuilder {
+    /**
+     * Adds a beneficiary for a deleted account.
+     * @param deletedAccountID the deleted account ID
+     * @param beneficiaryForDeletedAccount the beneficiary account ID
+     * @return the builder
+     */
+    @NonNull
+    CryptoDeleteRecordBuilder addBeneficiaryForDeletedAccount(
+            @NonNull final AccountID deletedAccountID, @NonNull final AccountID beneficiaryForDeletedAccount);
+
+    /**
+     * Gets the deleted account beneficiaries.
+     * @return the deleted account beneficiaries
+     */
+    // TODO: Not sure if this should be here or in SingleTranssactionRecordBuilder
```

Comment:
I lean towards having it here for now because this isn't something that should be generally available to all handlers, but it's a good question.